composer/core/time.py

```diff
@@ -391,37 +407,48 @@ def from_timestring(cls, timestring: str) -> Time:
 class Timestamp(Serializable):
     """Timestamp represents a snapshot of the current training progress.
 
-    The timestamp measures training progress in terms of epochs, batches, samples, tokens, and wall clock time.
+    The timestamp measures training progress in terms of iterations, epochs, batches, samples, tokens, and wall clock time.
     Timestamps are not updated in-place.
 
     See the :doc:`Time Guide </trainer/time>` for more details on tracking time during training.
 
     Args:
+        iteration (int | Time[int], optional): The iteration.
         epoch (int | Time[int], optional): The epoch.
         batch (int | Time[int], optional): the batch.
         sample (int | Time[int], optional): The sample.
         token (int | Time[int], optional): The token.
+        epoch_in_iteration (int | Time[int], optional): The epoch in the iteration.
         batch_in_epoch (int | Time[int], optional): The batch in the epoch.
         sample_in_epoch (int | Time[int], optional): The sample in the epoch.
         token_in_epoch (int | Time[int], optional): The token in the epoch.
         total_wct (datetime.timedelta, optional): The total wall-clock duration.
+        iteration_wct (datetime.timedelta, optional): The wall-clock duration of the last iteration.
         epoch_wct (datetime.timedelta, optional): The wall-clock duration of the last epoch.
         batch_wct (datetime.timedelta, optional): The wall-clock duration of the last batch.
     """
 
     def __init__(
         self,
+        iteration: Union[int, Time[int]] = 0,
         epoch: Union[int, Time[int]] = 0,
         batch: Union[int, Time[int]] = 0,
         sample: Union[int, Time[int]] = 0,
         token: Union[int, Time[int]] = 0,
+        epoch_in_iteration: Union[int, Time[int]] = 0,
         batch_in_epoch: Union[int, Time[int]] = 0,
         sample_in_epoch: Union[int, Time[int]] = 0,
         token_in_epoch: Union[int, Time[int]] = 0,
         total_wct: Optional[datetime.timedelta] = None,
+        iteration_wct: Optional[datetime.timedelta] = None,
         epoch_wct: Optional[datetime.timedelta] = None,
         batch_wct: Optional[datetime.timedelta] = None,
     ):
+        iteration = Time.from_input(iteration, TimeUnit.ITERATION)
+        if iteration.unit != TimeUnit.ITERATION:
```

Comment:
just for my understanding, when would this check not evaluate to True?

Reply:
You can pass in `iteration=Time(5, TimeUnit.EPOCH)`